File: Portfolio.Web.Api/Model/Investment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Portfolio.Web.Api.Model
{
    public class Investment
    {
        public long AccountId { get; set; }
        public long InvestmentId { get; set; }
        public string Name { get; set; }
        public decimal Quantity { get; set; }
        public decimal Cost { get; set; }

        public int StockId { get; set; }
        public string Ticker { get; set; }
        public decimal Price { get; set; }
        public decimal Dividend { get; set; }
        public double Minutes { get; set; }
        public bool Reload { get; set; }
        public string Tags { set; get; }

        public List<StockDividend> Dividends { get; set; }

        public Investment()
        {
            this.Dividends = new List<StockDividend>();
        }

        public Investment(Portfolio.Data.Model.Investment obj) : this() { this.Hydrate(obj); }

        public void Hydrate(Portfolio.Data.Model.Investment obj)
        {
            this.AccountId = obj.AccountId;
            this.InvestmentId = obj.InvestmentId;
            this.Name = obj.Name;
            this.Quantity = obj.Quantity;
            this.Cost = obj.Cost;
            if (obj.Stock != null)
            {
                this.Hydrate(obj.Stock);
            }
        }

        public void Hydrate(Portfolio.Data.Model.Stock obj)
        {
            this.StockId = obj.StockId;
            this.Name = obj.Name;
            this.Ticker = obj.Ticker;
            this.Price = obj.Price;
            this.Dividend = obj.Dividend;
            this.Minutes = obj.Minutes;
            this.Reload = obj.Reload;

            foreach (var item in obj.Dividends)
            {
                this.Dividends.Add(new StockDividend(item));
            }
        }
    }    
}

File: Portfolio.Web.App/app/models/Investment.js

c3o.Core.Data.InvestmentResponse = function (data) {
	var self = this;
	if (data) {
		this.accountId = data.accountId;
		this.investmentId = data.investmentId;
		this.name = data.name;
		this.quantity = data.quantity;
		//this.cost = data.cost;
		this.stockId = data.stockId;
		this.ticker = data.ticker;
		//this.price = data.price;
		//this.dividend = data.dividend;
		this.tags = data.tags;
	};
};


/////////////////////////////////////////////////////////////
// Investment Class
/////////////////////////////////////////////////////////////
c3o.Core.Data.Investment = function (data) {
	var self = this;

	if (data) {
		_.extend(this, data);

		// Transform Account Investments
		if (this.dividends != null && this.dividends != undefined) {
			this.dividends = $.map(this.dividends, function (item, i) {
				return new c3o.Core.Data.Dividend(item, self);
			});
		}
	};
};


// Investment class methods
c3o.Core.Data.Investment.prototype = {

	Price: function () {
			return this.price * this.quantity;
	},
	Income: function () {
		return (this.price * this.quantity * (this.dividend / 100));
	},


	//// Merge Additional product data into existing object
	//Merge: function (investment) {
	//	_.extend(this, investment);
	//	//this.summary = product.summary;
	//	//this.description = product.description;
	//	//this.content = product.content;
	//	//this.mediaObjects = product.mediaObjects;
	//}

	////get Ordered() { return this.orders && this.orders.length > 0; },
	////get Price() { return _.sum(this.ticketTypes, function (obj) { return obj.Price; }); },
};